src/metabase/driver/presto.clj

```diff
@@ -61,28 +66,32 @@
           (parser value))))))
 
 (defn- fetch-presto-results! [details {prev-columns :columns, prev-rows :rows} uri]
-  (let [{{:keys [columns data nextUri error]} :body} (http/get uri (assoc (details->request details) :as :json))]
-    (when error
-      (throw (ex-info (or (:message error) "Error running query.") error)))
-    (let [rows    (parse-presto-results columns data)
-          results {:columns (or columns prev-columns)
-                   :rows    (vec (concat prev-rows rows))}]
-      (if (nil? nextUri)
-        results
-        (do (Thread/sleep 100) ; Might not be the best way, but the pattern is that we poll Presto at intervals
-            (fetch-presto-results! details results nextUri))))))
+  (ssh/with-ssh-tunnel details
+    (fn fetch-presto-results-worker [details-with-tunnel]
```

Comment:
are these anonymous functions being used recursively? I think naming the functions here sort of implies that they are, it might be clearer if you just omit the name